test/Core.ts

```diff
@@ -384,45 +386,110 @@ export class Foo {
+   * ${importLine}
+   * 
+   * const bar = 1
    *
    * @since 0.0.1
    */
   public bar(): string {
     return 'bar'
   }
-}
-          `
-        })
-        const capabilities = makeCapabilities(state)
-
-        assertRight(await Core.main(capabilities)(), (result) => {
-          const exampleFileName = `${[...process.cwd().split('/'), 'src', 'utils', 'foo.ts'].join(
-            '-'
-          )}-Foo-method-bar-0.ts`
-          const exampleFilePath = `${process.cwd()}/docs/examples/${exampleFileName}`
-          const importFromRegex = new RegExp("from '../../src'")
+                            }`
+          })
+
+        /* By using the Array monad we generate a couple of PackageImport permutations
+         */
+        const packageImportPermutations: PackageImport[] = pipe(
+          A.bindTo('quotes')(['single' as const, 'double' as const]),
+          A.bind('scope', () => [O.some('my-org'), O.none]),
+          A.bind('name', () => ['my-package']),
+          A.bind('packageName', () =>
+            pipe(
+              A.bindTo('scope')([O.some('my-org'), O.none]),
+              A.bind('name', () => ['my-package'])
+            )
+          ),
+          A.bind('lib', () => [true, false]),
+          A.bind('path', () => [O.some('seg1'), O.some('seg1/seg2'), O.none])
+        )
 
-          assert.strictEqual(result, undefined)
-          assert.strictEqual(command, 'ts-node')
-          assert.strictEqual(executablePath.includes('docs/examples/index.ts'), true)
-          assert.strictEqual(importFromRegex.test(fileSystemState[exampleFilePath]), true)
-        })
+        test.each(
+          pipe(
+            packageImportPermutations,
+            A.map((packageImport) => [stringifyPackageImport(packageImport), packageImport])
+          )
+        )(
+          'should replace imports like %s from the project with the local source folder',
+          async (packageImportStr, { path, packageName }) => {
+            const state = mkState(packageName, `import * as foo from ${packageImportStr}`)
+            const capabilities = makeCapabilities(state)
+            assertRight(await Core.main(capabilities)(), (result) => {
+              const exampleFileName = `${[...process.cwd().split('/'), 'src', 'utils', 'foo.ts'].join(
+                '-'
+              )}-Foo-method-bar-0.ts`
+              const exampleFilePath = `${process.cwd()}/docs/examples/${exampleFileName}`
+              const importPath =
+                '../../src' +
+                O.fold(
+                  () => '',
+                  (str) => `/${str}`
+                )(path)
+
+              assert.strictEqual(result, undefined)
+              assert.strictEqual(command, 'ts-node')
+              assert.strictEqual(executablePath.includes('docs/examples/index.ts'), true)
+              assert.strictEqual(
+                fileSystemState[exampleFilePath],
+                `import * as foo from '${importPath}'
+
+const bar = 1
+`
```

Comment:
Generally I'm not so happy with those multi line strings. But testing against the real expected output strings (instead of booleans) produces much better outputs/diffs when tests fail.

Reply:
Agreed - even if the tests are ugly, a rich diff in the case of an error (especially in the CI environment) is preferable.

Reply:
Maybe in the future using something like
```ts
const x = [
  'first line',
  'second line'
].join('\n')
```

That's annoying to edit, but at least there's proper indentation.